tests/Unit/Support/PathTest.php

```diff
@@ -0,0 +1,184 @@
+use Tests\Psr4\Concerns\SystemConcern;
+
+class PathTest extends TestCase
+{
+    use SystemConcern, PhpunitConcern;
+
+    /**
+     * @test
+     */
+    public function shouldGetEmptyPath()
+    {
+        // given
+        $path = new Path([]);
+        // when
+        $asString = $path->toString();
+        // then
+        $this->assertSame("", $asString);
+    }
+
+    /**
+     * @test
+     */
+    public function shouldIgnoreEmptyChildren()
+    {
+        // given
+        $path = new Path(["string", "", "", "string"]);
+        // when
+        $asString = $path->toString();
+        // then
+        $this->assertSameWindows("string\string", $asString);
+        $this->assertSameUnix("string/string", $asString);
+    }
+
+    /**
+     * @test
+     * @dataProvider fileNames
+     */
+    public function shouldGetSingleFile(string $filename)
+    {
+        // given
+        $path = new Path([$filename]);
+        // when
+        $asString = $path->toString();
+        // then
+        $this->assertSame("file.txt", $asString);
+    }
+
+    public function fileNames(): array
+    {
+        return [["file.txt"], ["file.txt/"], ["file.txt\\"]];
+    }
+
+    /**
+     * @test
+     * @dataProvider pathPieces
+     */
+    public function shouldGetManyFiles(array $pathPieces)
+    {
+        // given
+        $path = new Path($pathPieces);
+        // when
+        $asString = $path->toString();
+        // then
+        $this->assertSameWindows('first\second\third\file.txt', $asString);
+        $this->assertSameUnix("first/second/third/file.txt", $asString);
+    }
+
+    public function pathPieces(): array
+    {
+        return [
+            [["first", "second", "third", "file.txt"]],
+
+            [["first", "/second", "/third", "/file.txt"]],
+            [["first", "\second", '\third', '\file.txt']],
+
+            [["first/", "second/", "third/", "file.txt"]],
+            [["first\\", "second\\", "third\\", "file.txt"]],
+
+            [["first/", "/second/", "/third/", "/file.txt"]],
+            [["first\\", "\\second\\", '\\third\\', '\\file.txt']],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider paths
+     */
+    public function shouldRepresentPath(string $stringPath)
+    {
+        // given
+        $path = Path::of($stringPath);
+        // when
+        $asString = $path->toString();
+        // then
+        $this->assertSameWindows('one\two\three\file.txt', $asString);
+        $this->assertSameUnix("one/two/three/file.txt", $asString);
+    }
+
+    public function paths(): array
+    {
+        return [['one\two\three\file.txt'], ["one/two/three/file.txt"]];
+    }
+
+    /**
+     * @test
+     * @dataProvider children
+     */
+    public function shouldAppendPath(Path $path, string $appendant)
+    {
+        // when
+        $childPath = $path->append($appendant);
+        // then
+        $this->assertSameWindows('uno\dos\tres', $childPath->toString());
```

Comment:
Please consider making `DIRECTORY_SEPARATOR` injectable for the `Path` class. That way unit tests are going to be system agnostic.

Reply:
Aren't they now with `assertSameWindows()` and `assertSameUnix()`?

Reply:
However, to be able to test windows behavior you need to run tests on windows VM. You can't be sure this code works correctly with windows when running tests on a unix-based system.

Reply:
@budziam Is there a feature request to build Windows paths on Unix and *vice-versa*? Because if there isn't, there's no way to test it. And to test it, it would mean to require that functionallity. However it seems off, doesn't it?

Perhaps to test it both on windows and unix, we can employ the same strategy we use for testing both php8 and php7, which are github actions. Would it be hard to run the tests on windows docker image?

> Please consider making `DIRECTORY_SEPARATOR` injectable for the `Path` class. That way unit tests are going to be system agnostic.

Tests would become more flexible, but the implementation would become less cohesive.

Reply:
CPU costs money and shop sms was never intended to be run on windows instances. Running tests against windows VMs is not an acceptable approach.

Reply:
Github Actions cost money?